arangod/IResearch/IResearchFeature.cpp

```diff
@@ -114,50 +115,42 @@ class IResearchLogTopic final : public LogTopic {
     setIResearchLogLevel(kDefaultLevel);
   }
 
-  virtual void setLogLevel(LogLevel level) override {
+  void setLogLevel(LogLevel level) final {
     LogTopic::setLogLevel(level);
     setIResearchLogLevel(level);
   }
 
  private:
   static constexpr LogLevel kDefaultLevel = LogLevel::INFO;
 
-  using irsLogLevelType = std::underlying_type_t<irs::logger::level_t>;
-  using arangoLogLevelType = std::underlying_type_t<LogLevel>;
-
-  static_assert(static_cast<irsLogLevelType>(irs::logger::IRL_FATAL) ==
-                        static_cast<arangoLogLevelType>(LogLevel::FATAL) - 1 &&
-                    static_cast<irsLogLevelType>(irs::logger::IRL_ERROR) ==
-                        static_cast<arangoLogLevelType>(LogLevel::ERR) - 1 &&
-                    static_cast<irsLogLevelType>(irs::logger::IRL_WARN) ==
-                        static_cast<arangoLogLevelType>(LogLevel::WARN) - 1 &&
-                    static_cast<irsLogLevelType>(irs::logger::IRL_INFO) ==
-                        static_cast<arangoLogLevelType>(LogLevel::INFO) - 1 &&
-                    static_cast<irsLogLevelType>(irs::logger::IRL_DEBUG) ==
-                        static_cast<arangoLogLevelType>(LogLevel::DEBUG) - 1 &&
-                    static_cast<irsLogLevelType>(irs::logger::IRL_TRACE) ==
-                        static_cast<arangoLogLevelType>(LogLevel::TRACE) - 1,
-                "inconsistent log level mapping");
-
-  static void log_appender(void* context, char const* function,
-                           char const* file, int line,
-                           irs::logger::level_t level, char const* message,
-                           size_t message_len);
-  static void setIResearchLogLevel(LogLevel level) {
-    if (level == LogLevel::DEFAULT) {
-      level = kDefaultLevel;
-    }
+  static void setIResearchLogLevel(LogLevel level);
+};
 
-    auto irsLevel = static_cast<irs::logger::level_t>(
-        static_cast<arangoLogLevelType>(level) - 1);  // -1 for DEFAULT
+static IResearchLogTopic LIBIRESEARCH("libiresearch");
 
-    irsLevel = std::max(irsLevel, irs::logger::IRL_FATAL);
-    irsLevel = std::min(irsLevel, irs::logger::IRL_TRACE);
-    irs::logger::output_le(irsLevel, log_appender, nullptr);
-  }
+template<LogLevel Level>
+static void log(irs::SourceLocation&& source, std::string_view message) {
+  Logger::log("9afd3", source.func.data(), source.file.data(),
+              static_cast<int>(source.line), Level, LIBIRESEARCH.id(), message);
+}
+
+static constexpr irs::log::Callback kLogs[6] = {
```

Comment:
std::array<Callback, 6> ?

Reply:
At least to have size and do not write constant "6" several times

Reply:
ok, but later or if this jenkins run will failed

Reply:
Fixed